src/main/java/com/cloudbees/jenkins/support/impl/GCLogs.java

```diff
@@ -34,6 +35,14 @@
 
     static final String GCLOGS_ROTATION_SWITCH = "-XX:+UseGCLogFileRotation";
 
+    private static final String GCLOGS_RETENTION_PROPERTY = GCLogs.class.getCanonicalName() + ".retention";
+
+    /**
+     * How many days of garbage collector log files should be included in the bundle. 
+     * By default {@code 3} days. Any value less or equals to {@code 0} disables the retention.
+     */
+    private static final Integer GCLOGS_RETENTION_DAYS = Math.max(0, Integer.getInteger(GCLOGS_RETENTION_PROPERTY, 2));
```

Comment:
Typically, reviewing data over a 24 hour period will provide enough information about the health of a JVM, However, Alex brings up a good point here that a slow memory leak pattern, while not typical, may not present itself over a 2-3 day period. Therefore, my opinion is to increase this to 5 days. If you aren't able to see a visible increase in pattern over 5 days, then it's time to dust off MAT.

Reply:
Maybe it can correlate how often a support bundle is created. If we need older GC logs then we can ask for the previous support bundles.
Another option is to set a 5 day default and add another option like "Extended GC" for performance issues.

Reply:
I agree that 2 days might be too short. 2 weeks sounds really long and defeats the purpose of this fix. What we want is having glance at the GC activity with enough data to identify a problem. If a memory leak is slow so that it can't be detected within a few days, we could argue about the impact / significance of the memory leak on the instance... If more historical data is needed, it is not lost and could be requested.

Reply:
Raised to 5 days.